File: C#/Logic/LinearSearch1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using C_.Class;

namespace C_.Logic
{
    internal class LinearSearch1 : LogicSearch
    {
        /// <summary>
        /// 線形探索法（リニアサーチ）
        /// </summary>
        public override void Search()
        {
            Person[] person = new Person[]
            {
                new Person("Ann", 18),
                new Person("Rolla", 19),
                new Person("Nancy", 16),
                new Person("Eluza", 17),
                new Person("Juliet", 18),
                new Person("Candy", 16)
            };

            int personLength = person.Length;
            string keyName = "Candy";
            int i = 0;
            while (i < personLength && keyName != person[i].Name)
            {
                i++;
            }

            if (i < personLength)
            {
                Console.WriteLine(person[i].Name + " " + person[i].Age);
            }
            else
            {
                Console.WriteLine("見つかりませんでした。");
            }
        }
    }
}

File: Java/Search/src/Logic/LinearSearch1.java
package Logic;

import Class.Person;
/**
 * 線形探索法（リニアサーチ）
 */
public class LinearSearch1 extends LogicSearch {
	public void Search() {
		Person[] p = new Person[6];
		p[0] = new Person("Ann", 18);
		p[1] = new Person("Rolla", 19);
		p[2] = new Person("Nancy", 16);
		p[3] = new Person("Eluza", 17);
		p[4] = new Person("Juliet", 18);
		p[5] = new Person("Machilda", 20);
		
		int count = p.length;
		String keyName = "Eluza";
		int index = 0;
		
		while(index < count && p[index].getName() != keyName) {
			index++;
		}
		
		if(index < count) {
			System.out.println(p[index].getName() + " " + p[index].getAge());
		} else {
			System.out.println("見つかりませんでした。");
		}
	}
}
